### PR TITLE
comments: add Comments Plugin

### PR DESCRIPTION
**Pull request by victorgcramos**
This PR adds the Comments plugin into our new architecture.

Includes:
- Comments, count, policy, timestamps and votes slices
- Comment Card, Comments List and Comments UI
- Comments filtering UI
- Flat mode and thread mode
- User vote highlighting
- Comments censoring UI

Ps: new/edit comments, censoring and comments voting weren't implemented yet. 

## UI Changes
image

### CODE REVIEW

**Comment by victorgcramos** [comment]
UI Changes checkpoint:
Screen Shot 2022-03-03 at 5 24 29 PM

**Review by tiagoalvesdulce — CHANGES_REQUESTED** [review]
Looks good overall. I left a few inline comments.

File: plugins-structure/packages/comments/src/comments/votes/votesSlice.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    builder
+      .addCase(fetchCommentsVotes.pending, (state) => {
+        state.status = "loading";
+      })
+      .addCase(fetchCommentsVotes.fulfilled, (state, action) => {
+        const { token, userid } = action.meta.arg;
+        const { votes } = action.payload;
+        // populate votes by comment ids
+        const formattedVotes = votes
+          .sort((a, b) => a.timestamp - b.timestamp) // Order by asc. timestamps
```
[the remark this answers is not in the document]
Comment (victorgcramos) [reply plugins-structure/packages/comments/src/comments/votes/votesSlice.js:40]: All vote actions are returned by backend. If some user casts two comments votes, the backend will return an array with these two actions, not necessarily ordered by timestamps. Here, I re-order it so we only count the most recent vote.

I'll write down a comment explaining it! Thanks.

File: plugins-structure/packages/comments/src/ui/Comments/utils.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+export function sortByNew(commentsById) {
+  return Object.values(commentsById).sort((a, b) => b.timestamp - a.timestamp);
+}
+
+export function sortByOld(commentsById) {
+  return Object.values(commentsById).sort((a, b) => a.timestamp - b.timestamp);
+}
+
+export function sortByTop(commentsById) {
```
[the remark this answers is not in the document]
Comment (victorgcramos) [reply plugins-structure/packages/comments/src/ui/Comments/utils.js:26]: Much better. Thanks!

File: plugins-structure/packages/comments/src/ui/Comments/utils.js [hunk 3]
```diff
@@ -1,29 +1,46 @@
...
+ * ];
+ * const threadSchema = getThreadSchema(comments, false);
+ * // { 0: [1], 1: [2, 3] }
+ * const flatThreadSchema = getThreadSchema(comments, true);
+ * // { 0: [1, 2, 3] }
+ * ```
+ * @param {Array} comments comments array
+ * @param {Boolean} isFlatMode
+ * @returns {Object} thread schema
+ */
```
Comment (tiagoalvesdulce) [inline plugins-structure/packages/comments/src/ui/Comments/utils.js:20]: 👍

File: plugins-structure/packages/comments/package.json [hunk 4]
```diff
@@ -18,13 +18,12 @@
...
+        "presets": [
+          "@babel/preset-react"
         ]
       }
-    },
-    "presets": [
-      "@babel/preset-env",
-      "@babel/preset-react"
-    ]
+    }
```
Comment (tiagoalvesdulce) [inline plugins-structure/packages/comments/package.json:26]: why do we need those presets here?

Comment (victorgcramos) [reply plugins-structure/packages/comments/package.json:26]: The react preset is required in order to make jest recognize the JSX syntax on our tests. If I remove it, I get the following error:

```bash
Test suite failed to run

    Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.
```

Any thoughts on this? I think adding the preset is enough to fix it.

Comment (tiagoalvesdulce) [reply plugins-structure/packages/comments/package.json]: yeah, it's fine to keep it.

**Review by tiagoalvesdulce — APPROVED** [review]
LGTM

**Pull request merged**